EventFilter/CastorRawToDigi/plugins/CastorRawToDigi.cc

```diff
@@ -72,37 +82,93 @@ void CastorRawToDigi::produce(edm::Event& e, const edm::EventSetup& es)
 
   CastorRawCollections colls;
   colls.castorCont=&castor;
+  colls.zdcCont=&zdc;
   if (unpackTTP_) colls.ttp=&ttp;
   colls.tpCont=&htp;
  
   // Step C: unpack all requested FEDs
+  const FEDRawData& fed722 = rawraw->FEDData(722);
+  const int fed722size = fed722.size();
+  const FEDRawData& fed693 = rawraw->FEDData(693);
+  const int fed693size = fed693.size();
   for (std::vector<int>::const_iterator i=fedUnpackList_.begin(); i!=fedUnpackList_.end(); i++) {
     const FEDRawData& fed = rawraw->FEDData(*i);
-    if (fed.size()==0) {
-      if (complainEmptyData_) {
-	edm::LogWarning("EmptyData") << "No data for FED " << *i;
-	report->addError(*i);
-      }
-    } else if (fed.size()<8*3) {
-      edm::LogWarning("EmptyData") << "Tiny data " << fed.size() << " for FED " << *i;
-      report->addError(*i);
-    } else {
-      try {
-		  if ( usingctdc_ ) { 
-			  ctdcunpacker_.unpack(fed,*readoutMap,colls, *report);
-		  } else {
-			  unpacker_.unpack(fed,*readoutMap,colls, *report);
+    //std::cout<<"Fed number "<<*i<<"is being worked on"<<std::endl;
+    if (*i == 693 && fed693size == 0 && fed722size != 0)
+      continue;
+    if (*i == 722 && fed722size == 0 && fed693size != 0)
+      continue;      
+      
+    if (*i!=693 && *i!=722)
+      {
+	if (fed.size()==0) 
+	  {
+	    if (complainEmptyData_) 
+	      {
+		edm::LogWarning("EmptyData") << "No data for FED " << *i;
+		report->addError(*i);
 	      }
-	      report->addUnpacked(*i);
-      } catch (cms::Exception& e) {
-	edm::LogWarning("Unpacking error") << e.what();
-	report->addError(*i);
-      } catch (...) {
-	edm::LogWarning("Unpacking exception");
-	report->addError(*i);
+	  } 
+	else if (fed.size()<8*3) 
+	  {
+	    edm::LogWarning("EmptyData") << "Tiny data " << fed.size() << " for FED " << *i;
+	    report->addError(*i);
+	  } 
+	else 
+	  {
+	    try 
+	      {
+		if ( usingctdc_ ) 
+		  { 
+		    ctdcunpacker_.unpack(fed,*readoutMap,colls, *report);
+		  } 
+		else 
+		  {
+		    unpacker_.unpack(fed,*readoutMap,colls, *report);
+		  }
+		report->addUnpacked(*i);
+	      } 
+	    catch (cms::Exception& e) 
+	      {
+		edm::LogWarning("Unpacking error") << e.what();
+		report->addError(*i);
+	      } catch (...) 
+	      {
+		edm::LogWarning("Unpacking exception");
+		report->addError(*i);
+	      }
+	  }
       }
-    }
-  }
+
+    if (*i==693 && unpackZDC_)
+      {
+	if (fed.size()==0)
+	  {
+	    edm::LogWarning("EmptyData") << "No data for FED "<< *i;
```

Comment:
please hide this and the other on L161 with an ``` if  (complainEmptyData_)  ``` as it is done for the rest of castor FEDs.
We can't accept this warning to show up in every event for most of the data that we take (most of it doesn't have ZDC).